app/src/main/java/com/ikarabulut/Rules/RuleSet.java

```diff
@@ -0,0 +1,17 @@
+package com.ikarabulut.Rules;
+
+public class RuleSet {
+
+    public boolean willBeAliveNextGeneration(int aliveNeighborsCount, boolean isAlive) {
+        return isFertileLand(aliveNeighborsCount, isAlive);
+    }
+
+
+    protected boolean isFertileLand(int aliveNeighborsCount, boolean isAlive) {
```

Comment:
The way it usually works with a non-abstract base class is that methods that must be implemented by sub-classes should throw an unchecked exception, something like `NotImplementedError`. Then DefaultRules would implement these default rules.

In that scenario, the DefaultRules class wouldn't be empty, and this implementation code would live in there.

Context: https://stackoverflow.com/questions/829136/what-is-the-standard-exception-to-throw-in-java-for-not-supported-implemented-op

cc: @riccjohn

Reply:
@ikarabulut you don't have to change it, but something to think about going forward.  I also remembered that John had a question about whether the empty implementation class for DefaultRules was the normal way to do it or not.